teku/src/test/java/tech/pegasys/teku/cli/options/DepositOptionsTest.java

```diff
@@ -148,4 +147,52 @@ public void shouldThrowErrorIfSnapshotPathProvidedWithBundleSnapshot() {
         .isInstanceOf(InvalidConfigurationException.class)
         .hasMessage("Use either custom deposit tree snapshot path or snapshot bundle");
   }
+
+  @ParameterizedTest
+  @MethodSource("getDepositSnapshotOptions")
+  public void shouldHandleSnapshotPathAndEnabledConfig(
+      final String[] args, final String expectedPath, final boolean expectedIsEnabled) {
+    final PowchainConfiguration config = getTekuConfigurationFromArguments(args).powchain();
+    assertThat(config.getDepositSnapshotPath()).isEqualTo(Optional.of(expectedPath));
+    assertThat(config.isDepositSnapshotEnabled()).isEqualTo(expectedIsEnabled);
+  }
+
+  public static Stream<Arguments> getDepositSnapshotOptions() {
+    final String[] emptyConfig = {"--network=mainnet"};
+    final String[] depositSnapshotEnabledSet = {
+      "--network=mainnet",
+      "--deposit-snapshot-enabled=true",
+      "--checkpoint-sync-url=http://checkpoint/path/"
+    };
+    final String[] depositSnapshotPathSet = {"--Xdeposit-snapshot=/some/path/"};
+    final String[] depositSnapshotPathSetAndCheckpointSync = {
+      "--Xdeposit-snapshot=/some/path/",
+      "--deposit-snapshot-enabled=false",
+      "--checkpoint-sync-url=http://checkpoint/path/"
+    };
+    final String[] depositSnapshotEnabledFalseAndSync = {
```

Comment:
NIT: Maybe we should name this `depositSnapshotDisabledAndSync`

`EnabledFalse` kinda twisted my brain :P